lib/universal/models/attachment.rb

```diff
@@ -24,6 +25,25 @@ module Attachment
         scope :for_comment, -> (c){where(tci: c)}
         scope :for_name, ->(n){where(name: n)}
         scope :recent, ->(){order_by(created_at: :desc)}
+        before_update :download_uploaded_file
+        after_update :move_uploaded_file
+        
+        def download_uploaded_file
+          if !self.temporary_comment_id.blank?
+            new_file_path = Rails.root.join('tmp', "#{self.file_filename}")
+            File.open(new_file_path, 'wb') do |f|
+              f << open("http:#{self.file.url.gsub(/hms\/(.*)attachment/, "hms/attachment")}").read
```

Comment:
is this `gsub` necessary?  the file url should be correct as on the before_update, the subject will not exist

Reply:
The URL before update is something like:
```
//commondatastorage.googleapis.com/hms/comment/5a1cc3dffb608d1155000001/attachment/5a1cc3dcfb608d1155000000/Icebox_Nov_15_2017__2_.xlsx
```
At this point the file is actually located here:
```
//commondatastorage.googleapis.com/hms/attachment/5a1cc3dcfb608d1155000000/Icebox_Nov_15_2017__2_.xlsx
```